src/main/java/edu/cornell/kfs/sys/web/CuJaxbProvider.java

```diff
@@ -0,0 +1,89 @@
+package edu.cornell.kfs.sys.web;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.util.Set;
+
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.ext.MessageBodyReader;
+import javax.ws.rs.ext.MessageBodyWriter;
+
+import org.apache.commons.lang3.StringUtils;
+
+import edu.cornell.kfs.sys.CUKFSConstants;
+import edu.cornell.kfs.sys.service.CUMarshalService;
+import edu.cornell.kfs.sys.service.impl.CUMarshalServiceImpl;
+import jakarta.xml.bind.JAXBException;
+import jakarta.xml.bind.annotation.XmlRootElement;
+
+/**
+ * Custom JAX-RS message reader/writer for handling JAXB-annotated POJOs.
+ * This class allows POJOs containing the "jakarta.xml" JAXB annotations
+ * to be usable by the Jersey 2.x framework, which normally only supports
+ * the equivalent "javax.xml" annotations.
+ * 
+ * TODO: Remove this class when we either upgrade to Jersey 3.x or fully migrate from JAX-RS to Spring MVC.
+ */
+public class CuJaxbProvider implements MessageBodyReader<Object>, MessageBodyWriter<Object> {
+
+    private static final Set<String> XML_MEDIA_TYPES = Set.of(
+            MediaType.TEXT_XML, MediaType.APPLICATION_XML, MediaType.APPLICATION_ATOM_XML,
+            MediaType.APPLICATION_SVG_XML, MediaType.APPLICATION_XHTML_XML);
+
+    private CUMarshalService cuMarshalService = new CUMarshalServiceImpl();
+
+    @Override
+    public boolean isReadable(Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType) {
+        return isXmlMediaType(mediaType) && isAnnotatedWithXmlRootElement(type);
+    }
+
+    @Override
+    public boolean isWriteable(Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType) {
+        return isXmlMediaType(mediaType) && isAnnotatedWithXmlRootElement(type);
+    }
+
+    private boolean isXmlMediaType(MediaType mediaType) {
+        String mediaTypeString = StringUtils.lowerCase(mediaType.toString());
+        if (StringUtils.contains(mediaTypeString, CUKFSConstants.SEMICOLON)) {
+            mediaTypeString = StringUtils.substringBefore(mediaTypeString, CUKFSConstants.SEMICOLON);
+        }
```

Comment:
The media type string might contain key-value pairs for other properties, especially for specifying the charset encoding. (For example, the full media type string could end up being `application/xml;charset=utf-8` instead of just `application/xml`; an example is also available in the stack trace snippet on the KFSPTS-26394 ticket.) To simplify the comparisons, this code truncates the charset part before checking the media type.

Reply:
to simplify this further, could the check just see if the mediaType contains "xml"?

Reply:
Ok, I'll look into simplifying this check. It would probably be best to look for the word "xml" between the slash and the semicolon.